### DISCUSSION TITLE
Validating Properties

### DISCUSSION THREAD

**Discussion by RealDotNetDave** [canton7/PropertyChanged.SourceGenerator#40] [Q&A]
Using this generator, how do I validate a property value? I don't see that is done and I don't see it in the unit tests either. The only thing I saw was this:

[    private void OnFirstNameChanged(string oldValue, string newValue)
    {
        // ...
    } ](url)

Do I have to create a method like this for every property?

Thanks!

**Comment by canton7** [comment]
There's nothing built-in to do validation.

I recommend using something like FluentValidation. You should be able to call it from `OnAnyPropertyChanged`.
